code/_onclick/hud/screen_objects.dm

```diff
@@ -666,6 +631,50 @@
 	icon_state = "mood5"
 	screen_loc = ui_mood
 
+/obj/screen/combattoggle
+	name = "toggle combat mode"
+	icon = 'icons/mob/screen_midnight.dmi'
+	icon_state = "combat_off"
+	screen_loc = ui_combat_toggle
+
+/obj/screen/combattoggle/New(loc, ...)
+	. = ..()
+	update_icon()
+
+/obj/screen/combattoggle/Click()
+	if(isliving(usr))
+		var/mob/living/owner = usr
+		owner.set_combat_mode(!owner.combat_mode)
+		update_icon()
+
+/obj/screen/combattoggle/update_icon_state()
+	. = ..()
+	var/mob/living/user = hud?.mymob
+	if(!istype(user) || !user.client)
+		return
+	icon_state = user.combat_mode ? "combat" : "combat_off" //Treats the combat_mode
+
+//Version of the combat toggle with the flashy overlay
+/obj/screen/combattoggle/flashy
+	///Mut appearance for flashy border
+	var/mutable_appearance/flashy
```

Comment:
This is something we have on tgmc:
```
/atom/movable/vis_obj
	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
```
From that we derive visual objects with less code pollution and the `/obj` children.
This could be substituted for one:
```
/atom/movable/vis_obj/screen
	icon = 'icons/mob/screen_gen.dmi'
	layer = HUD_LAYER
	plane = HUD_PLANE

/atom/movable/vis_obj/screen/flashy_border
	icon_state = "togglefull_flash"
	color = = "#C62727" //This should be a color define.
```
That way you have a singleton that gets applied to every screen object.
Check tgmc's `visual_objects.dm` for an example of what I mean.

Reply:
I'll check it out, some more explanation on discord regarding this might be nice

Reply:
determined this wont be as useful as we thought as my mutables will have client-determined colors and thus not singleton